File: python_sol/p4.py
# Project Euler #4
# Find the largest palindrome number that is the multiple of three-digit numbers

num = 999
palin = []

while num > 100:
    for i in range(100, num):
        prod = num*i
        test = str(prod)
        if test == test[::-1]:
            palin.append(int(test))
    num -= 1

print(max(palin))

File: ruby_sol/p4.rb
#!/usr/bin/ruby
# Find the largest palindrome made from the product of two 3-digit numbers

num = 999
palin = []

while num > 100
  (100..num).each do |i|
      prod = num*i
      test = prod.to_s
      if test == test.reverse
        palin.push(test.to_i)
      end
  end
  num -= 1
end

puts palin.max
